src/blueprints/tv2/timeline-object-factories/interfaces/tv2-video-mixer-timeline-object-factory.ts

```diff
@@ -6,9 +6,9 @@ import { Tv2BlueprintConfiguration } from '../../value-objects/tv2-blueprint-con
 import { Tv2BlueprintTimelineObject } from '../../value-objects/tv2-metadata'
 
 export interface Tv2VideoMixerTimelineObjectFactory {
-  createProgramTimelineObject(sourceInput: number, enable: TimelineEnable): TimelineObject
+  createProgramTimelineObject(sourceInput: number, enable: TimelineEnable, id?: string): TimelineObject
```

Comment:
Is the optional `id` parameter really necessary?
Also, having the `id` parameter as  the first parameter would be nice and would be consistent with the rest of the codebase. That however makes it difficult to make it optional.

Reply:
The `id` parameters are required and are rearranged to be the first parameter of the methods.